src/modules/job-exec/exec_config.c

```diff
@@ -216,6 +242,69 @@ int config_init (flux_t *h, int argc, char **argv)
               exec_service,
               exec_service_override ? "enabled" : "disabled",
               sdexec_properties ? "set" : "not set");
+}
+
+/*  Initialize common configurations for use by job-exec exec modules.
+ */
+int config_init (flux_t *h, int argc, char **argv)
+{
+    const flux_conf_t *conf;
+
+    /* Per trws comment in 97421e88987535260b10d6a19551cea625f26ce4
+     *
+     * The musl libc loader doesn't actually unload objects on
+     * dlclose, so a subsequent dlopen doesn't re-clear globals and
+     * similar.
+     *
+     * So we must re-initialize globals on each config_init().
+     */
+    default_cwd = "/tmp";
+    default_job_shell = NULL;
+    default_job_shell_set_cmdline = false;
+    flux_imp_path = NULL;
+    flux_imp_path_set_cmdline = false;
+    exec_service = "rexec";
+    exec_service_set_cmdline = false;
+    exec_service_override = 0;
+    sdexec_properties = NULL;
+
+    if (!(conf = flux_get_conf (h))) {
+        flux_log_error (h, "error retrieving flux conf");
+        return -1;
+    }
+
+    if (conf_parse (h, conf) < 0)
+        return -1;
+
+    if (argv && argc) {
+        /* Finally, override values on cmdline */
+        for (int i = 0; i < argc; i++) {
+            if (strstarts (argv[i], "job-shell=")) {
+                default_job_shell = argv[i]+10;
+                default_job_shell_set_cmdline = true;
+            }
+            else if (strstarts (argv[i], "imp=")) {
+                flux_imp_path = argv[i]+4;
+                flux_imp_path_set_cmdline = true;
+            }
+            else if (strstarts (argv[i], "service=")) {
+                exec_service = argv[i]+8;
+                exec_service_set_cmdline = true;
+            }
+        }
+    }
+
+    conf_debug (h);
+    return 0;
+}
+
+int config_reload (flux_t *h, const flux_conf_t *conf)
+{
+    if (conf_parse (h, conf) < 0)
+        return -1;
+
+    conf_debug (h);
```

Comment:
This function should not print anything to the logs if the exec configuration has not changed.

Idea:  add an optional "impl" callback that returns a JSON object that is included in the module stats result?  Then the values that are actually in use could be queried by tests directly rather than relying on the logs.

Reply:
> Idea: add an optional "impl" callback that returns a JSON object that is included in the module stats result? Then the values that are actually in use could be queried by tests directly rather than relying on the logs.

I like the idea of adding this stuff to module stats, but not sure what you mean by an optional callback?  Perhaps you were mixing two ideas together?

Reply:
Oh I was just suggesting a new callback because the "implementation" probably wouldn't register the stats callback - that would happen in the main.  A callback in the main would need some implementation dependent function to call to get the stats info.